### ISSUE TITLE
Players go flying upwards on steep slopes

### ISSUE THREAD

**Issue by Gamepro5** [Whimfoome/godot-FirstPersonStarter#6]
**Expected behavior:** Since it isn't part of the floor angle, players slide down it as if it wasn't there with gravity.

**Current behavior:** Players treat the steep 15 degree slope as a catapult and go flying.

[Video of issue](https://youtu.be/_0m0wdVpDbk)

You don't have to fix these but I've been trying to do a good movement system in Godot for 4 weeks and would really like some help. There is another small issue that isn't related to this issue I shall share latter.

**Comment by Whimfoome** [comment]
The problem occurs because we get only the Y-axis of the velocity remainder. If we get the horizontal remainder, this problem will be fixed, but sliding on slopes will be broken
```
# Before
velocity.y = move_and_slide_with_snap(velocity, _snap, FLOOR_NORMAL, true, 4, deg2rad(floor_max_angle)).y

# After
velocity = move_and_slide_with_snap(velocity, _snap, FLOOR_NORMAL, true, 4, deg2rad(floor_max_angle))
```

**Comment by Whimfoome** [comment]
I may be wrong. The issue may come not from the velocity remainder, but from the unexpected behavior of the **floor_max_angle** in the **move_and_slide_with_slope** function

**Comment by Whimfoome** [comment]
Fixed in [52b0034](https://github.com/Whimfoome/godot-FirstPersonStarter/commit/52b0034d042d24257dbec4156f1e0a5e3419890c) with **is_on_wall()** solution from your pull request #7
